fastapi_route_logger_middleware/__init__.py

```diff
@@ -27,9 +28,18 @@ async def dispatch(self, request: Request, call_next: Callable) -> Response:
 
     def _should_route_be_skipped(self, request: Request) -> bool:
         return any(
-            [path for path in self._skip_routes if request.url.path.startswith(path)]
+            filter(self.__skip_route_match_filter(request.url.path), self._skip_routes)
         )
 
+    def __skip_route_match_filter(self, path: str):
+        def should_skip(skipped_route: str) -> bool:
+            try:
+                return re.compile(skipped_route).match(path) is not None
+            except:
+                return path.startswith(path)
```

Comment:
This should raise an exception. In fact, it should probably not catch the exception at all.

Reply:
Not all valid URL strings compile to regexs. If you want to maintain backwards compatibility, and also support regexs in the same `skip_routes` keyword, then this is necessary. You were a bit unclear in which of the directions I proposed you preferred, so I figured you wanted me to overload the existing `skip_routes` with the support.

Reply:
Honestly, I didn't care which way it went, backwards compatibility was the only requirement. 

I have two issues with this approach. First, there is no feedback that the regex is invalid. If that was an accident, as in added unit test, finding the problem will be difficult as no feedback on the problem is provided. Second, the code is assuming that every exception is because the regex is invalid.

I am fine with adding a new argument (e.g. skip_routes_regex or a better name) or fixing this issue. But these issues do need to be fixed.